x/ethbridge/keeper/msg_server_test.go

```diff
@@ -0,0 +1,119 @@
+package keeper_test
+
+import (
+	"testing"
+
+	adminTypes "github.com/Sifchain/sifnode/x/admin/types"
+	ethbriddgeKeeper "github.com/Sifchain/sifnode/x/ethbridge/keeper"
+	"github.com/Sifchain/sifnode/x/ethbridge/test"
+	"github.com/Sifchain/sifnode/x/ethbridge/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMsgServer_Lock_No_Pauser_Set(t *testing.T) {
+	ctx, app := test.CreateSimulatorApp(false)
+	addresses, _ := test.CreateTestAddrs(2)
+	admin := addresses[0]
+	// nonAdmin := addresses[1]
+	msg := types.NewMsgLock(1, admin, ethereumSender, amount, "stake", amount)
+	coins := sdk.NewCoins(sdk.NewCoin("stake", amount), sdk.NewCoin(types.CethSymbol, amount))
+	_ = app.BankKeeper.MintCoins(ctx, types.ModuleName, coins)
+	_ = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, admin, coins)
+	app.AdminKeeper.SetAdminAccount(ctx, &adminTypes.AdminAccount{
+		AdminType:    adminTypes.AdminType_ETHBRIDGE,
+		AdminAddress: admin.String(),
+	})
+	msgServer := ethbriddgeKeeper.NewMsgServerImpl(app.EthbridgeKeeper)
+
+	_, err := msgServer.Lock(sdk.WrapSDKContext(ctx), &msg)
+	require.NoError(t, err)
+}
+
+func TestMsgServer_Lock(t *testing.T) {
+	ctx, app := test.CreateSimulatorApp(false)
+	addresses, _ := test.CreateTestAddrs(2)
+	admin := addresses[0]
+	nonAdmin := addresses[1]
+	msg := types.NewMsgLock(1, admin, ethereumSender, amount, "stake", amount)
+	coins := sdk.NewCoins(sdk.NewCoin("stake", amount), sdk.NewCoin(types.CethSymbol, amount))
+	_ = app.BankKeeper.MintCoins(ctx, types.ModuleName, coins)
+	_ = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, admin, coins)
+	app.AdminKeeper.SetAdminAccount(ctx, &adminTypes.AdminAccount{
+		AdminType:    adminTypes.AdminType_ETHBRIDGE,
+		AdminAddress: admin.String(),
+	})
+	msgPauseNonAdmin := types.MsgPauser{
+		Signer:   nonAdmin.String(),
+		IsPaused: true,
+	}
+	msgPause := types.MsgPauser{
+		Signer:   admin.String(),
+		IsPaused: true,
+	}
+	msgUnPause := types.MsgPauser{
+		Signer:   admin.String(),
+		IsPaused: false,
+	}
```

Comment:
Should we test setting pause to false when its currently not paused to verify that the corner case works? So not paused, set pause to false, lock and burn still work.